File: src/SnapToTable.Application/DTOs/RecipeDto.cs
namespace SnapToTable.Application.DTOs;
public record RecipeDto(
    Guid Id,
    DateTime CreatedAt,
    Guid RecipeAnalysisId,
    string Name,
    string Category,
    TimeSpan? PrepTime,
    TimeSpan? CookTime,
    TimeSpan? AdditionalTime,
    int? Servings,
    IReadOnlyList<string> Ingredients,
    IReadOnlyList<string> Directions,
    IReadOnlyList<string> Notes
);

File: client/snap-to-table/features/recipes/api/dto/recipeDto.ts
export interface RecipeDto {
    id: string;
    createdAt: string;
    recipeAnalysisId: string;
    name: string;
    category: string;
    url: string | null;
    prepTime: string | null;
    cookTime: string | null;
    additionalTime: string | null;
    servings: number | null;
    
    readonly ingredients: readonly string[];
    readonly directions: readonly string[];
    readonly notes: readonly string[];
}